internal/platform/constants.go

```diff
@@ -0,0 +1,57 @@
+package platform
+
+import "fmt"
+
+const (
+	LinuxTag = uint32(iota) << platformShift
+)
+
+const (
+	platformMax   = 0xf
+	platformShift = 28
+	platformMask  = platformMax << platformShift
+)
+
+func tagForPlatform(platform string) (uint32, error) {
+	switch platform {
+	case Linux:
+		return LinuxTag, nil
+	default:
+		return 0, fmt.Errorf("unrecognized platform: %s", platform)
+	}
+}
+
+func platformForConstant(c uint32) string {
+	tag := uint32(c & platformMask)
+	switch tag {
+	case LinuxTag:
+		return Linux
+	default:
+		return ""
+	}
+}
+
+// Encode a platform and a value into a tagged constant.
+//
+// The platform tag is stored in the 4 most significant bits. The tags value is
```

Comment:
What is meant by the `tags` value? The platform constant I assume is LinuxTag in this case? Ack on `LinuxTag` being zero for backwards compat, but the 'one less than ..' throws me off. Maybe the comment just hasn't evolved with the rest of the code.